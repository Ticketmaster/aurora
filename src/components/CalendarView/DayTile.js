import styled from "styled-components";
import classnames from "classnames";

import { constants, colors } from "../../theme";

const DAY_TILE_CLASS = "day-tile";
const DAY_TILE_NO_BORDER_RADIUS_CLASS = `${DAY_TILE_CLASS}--no-border-radius`;

const DayTile = styled.div.attrs({
  className: ({ noBorderRadius }) =>
    classnames(DAY_TILE_CLASS, {
      [DAY_TILE_NO_BORDER_RADIUS_CLASS]: noBorderRadius
    })
})`
  position: relative;
  display: flex;
  flex-flow: column nowrap;
  height: 100%;
  min-height: 192px;
  background-color: ${colors.white.base};
  border: solid 1px ${colors.onyx.muted};
  border-radius: ${constants.borderRadius.small};
  overflow: hidden;

  &.${DAY_TILE_NO_BORDER_RADIUS_CLASS} {
    border-radius: 0;
  }
`;

DayTile.Header = styled.header`
  z-index: 1;
  position: absolute;
  top: 0;
  left: 0;
  right: 30px;
  display: flex;
  flex-flow: row nowrap;
  pointer-events: none;
`;

export default DayTile;
