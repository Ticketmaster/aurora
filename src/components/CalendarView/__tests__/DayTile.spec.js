import React from "react";
import renderer from "react-test-renderer";
import DayTile from "../DayTile";

const DEFAULT_PROPS = {
  dateLabel: "10",
  title: "Henderson, NV â€” Sunset Station Outdoor Amphitheater",
  subTitle: "On Sale: Fri, 08/17/18 10:00 AM",
  ctaButtons: [
    <DayTile.Button>2:00 pm</DayTile.Button>,
    <DayTile.Button>6:00 pm</DayTile.Button>
  ],
  moreButton: <DayTile.MoreButton>+4 More Times</DayTile.MoreButton>,
  onOverflowClick: () => {},
  withOverflow: true
};

describe("DayTile", () => {
  it("should render without errors", () =>
    expect(
      renderComponent({
        ...DEFAULT_PROPS
      })
    ).toMatchSnapshot());

  function renderComponent(props = {}) {
    return renderer.create(<DayTile {...props} />).toJSON();
  }
});
