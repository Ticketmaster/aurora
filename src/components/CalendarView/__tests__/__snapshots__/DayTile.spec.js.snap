// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayTile should render without errors 1`] = `
.c4 {
  font-size: 16px;
  font-weight: 600;
  line-height: 1.5;
  text-transform: none;
  color: #C56BFF;
}

.c12 {
  font-size: 12px;
  font-weight: 400;
  line-height: 1.5;
  text-transform: none;
  color: rgba(38,38,38,1);
}

.c13 {
  font-size: 12px;
  font-weight: 400;
  line-height: 1.5;
  text-transform: none;
  color: rgba(38,38,38,0.6);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 30px;
  height: 30px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(255,255,255,1);
}

.c17 {
  font-weight: 600;
  font-size: 12px;
  line-height: 1.84;
  width: 100%;
  padding: 0 10px 0 10px;
  min-width: 60px;
  text-align: center;
  border-radius: 2px;
  cursor: pointer;
  color: rgba(255,255,255,1);
  background-color: #026cdf;
  border: 1px solid transparent;
  -webkit-transition: -webkit-transform 0.1s linear;
  -webkit-transition: transform 0.1s linear;
  transition: transform 0.1s linear;
  -webkit-transition: background-color 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: background-color 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
}

.c17:focus {
  outline: none;
  box-shadow: 0 0 5px 0 #026cdf;
}

.c17:hover {
  background-color: #0150a7;
}

.c17:active {
  -webkit-transform: scale(0.98,0.98) translate(0,1px);
  -ms-transform: scale(0.98,0.98) translate(0,1px);
  transform: scale(0.98,0.98) translate(0,1px);
  background-color: #013670;
}

.c17:disabled {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  color: rgba(255,255,255,1);
  background-color: #026cdf;
  border: 1px solid transparent;
  cursor: not-allowed;
  opacity: 0.2;
}

.c19 {
  font-weight: 600;
  font-size: 12px;
  line-height: 1.84;
  width: 100%;
  padding: 0 10px 0 10px;
  min-width: 60px;
  text-align: center;
  text-transform: capitalize;
  border-radius: 2px;
  cursor: pointer;
  color: #026cdf;
  background-color: transparent;
  border: 1px solid transparent;
  -webkit-transition: -webkit-transform 0.1s linear;
  -webkit-transition: transform 0.1s linear;
  transition: transform 0.1s linear;
  -webkit-transition: background-color 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: background-color 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
}

.c19:focus {
  outline: none;
  box-shadow: 0 0 5px 0 #026cdf;
}

.c19:hover {
  background-color: #ebf4fd;
}

.c19:active {
  -webkit-transform: scale(0.98,0.98) translate(0,1px);
  -ms-transform: scale(0.98,0.98) translate(0,1px);
  transform: scale(0.98,0.98) translate(0,1px);
  background-color: #d6e7fa;
}

.c19:disabled {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  color: #026cdf;
  background-color: transparent;
  border: 1px solid transparent;
  cursor: not-allowed;
  opacity: 0.2;
}

.c8 {
  height: 36px;
  width: 100%;
  min-width: 100px;
  font-weight: 600;
  text-align: center;
  border-radius: 2px;
  color: rgba(255,255,255,1);
  background-color: transparent;
  border: 1px solid rgba(255,255,255,1);
  cursor: pointer;
  text-transform: capitalize;
  font-size: 14px;
  min-width: 30px;
  max-width: 30px;
  height: 30px;
  width: 30px;
  border: none;
}

.c8:disabled {
  color: rgba(255,255,255,0.5);
  background-color: transparent;
  border: 1px solid rgba(255,255,255,0.5);
  cursor: not-allowed;
}

.c8:focus {
  outline: none;
}

.c8 > * {
  pointer-events: none;
}

.c5 {
  margin-left: auto;
}

.c7 {
  display: block;
  padding: 8px 8px 0;
  border: 0;
}

.c6:hover .c9 {
  color: rgba(2,108,223,1);
  fill: currentColor;
}

.c18 {
  border-top: 1px solid rgba(38,38,38,0.4);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  min-height: 190px;
  background-color: rgba(255,255,255,1);
  border: solid 1px rgba(38,38,38,0.4);
  border-radius: 2px;
  overflow: hidden;
}

.c1.day-tile--with-image {
  position: relative;
  min-height: 30px;
}

.c1.day-tile--highlighted {
  background-color: rgba(2,108,223,0.1);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c0.day-tile--with-image > .c2 {
  z-index: 1;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
}

.c10 {
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
}

.c0:not(.day-tile--with-image) > .c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  padding: 8px 8px 4px;
}

.c14 {
  margin-top: auto;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0 4px;
}

.c16 {
  width: calc(50% - 8px);
  margin: 0 4px 8px;
}

@media screen and (min-width:768px) {
  .c4 {
    font-size: 16px;
  }
}

@media screen and (min-width:1024px) {
  .c4 {
    font-size: 16px;
  }
}

@media screen and (min-width:768px) {
  .c12 {
    font-size: 12px;
  }
}

@media screen and (min-width:1024px) {
  .c12 {
    font-size: 12px;
  }
}

@media screen and (min-width:768px) {
  .c13 {
    font-size: 12px;
  }
}

@media screen and (min-width:1024px) {
  .c13 {
    font-size: 12px;
  }
}

<article
  className="c0 day-tile c1"
>
  <header
    className="c2"
  >
    <div
      className="c3"
    >
      <div
        className="text text--accent text--heliotrope c4"
        disabled={false}
        size={
          Object {
            "large": "kilo",
            "medium": "kilo",
            "small": "kilo",
          }
        }
      >
        10
      </div>
    </div>
    <div
      className="c5"
    >
      <button
        aria-label="More Info"
        className="c6 c7 c8"
        onClick={[Function]}
        role="button"
        size={30}
      >
        <svg
          className="c9"
          fill="rgba(38, 38, 38, 0.6)"
          height={21}
          viewBox="0 0 5 21"
          width="6"
        >
          <path
            d="M0 2.5C0 1.81.242 1.22.727.732A2.394 2.394 0 0 1 2.492 0a2.42 2.42 0 0 1 1.773.732C4.755 1.221 5 1.81 5 2.5s-.245 1.28-.735 1.768A2.42 2.42 0 0 1 2.492 5C1.8 5 1.212 4.756.727 4.268A2.418 2.418 0 0 1 0 2.5zm0 8.008c0-.692.242-1.283.727-1.773A2.39 2.39 0 0 1 2.492 8c.692 0 1.283.245 1.773.735S5 9.815 5 10.508a2.39 2.39 0 0 1-.735 1.765A2.43 2.43 0 0 1 2.492 13c-.692 0-1.28-.242-1.765-.727A2.403 2.403 0 0 1 0 10.508zM0 18.5c0-.69.242-1.28.727-1.768A2.394 2.394 0 0 1 2.492 16a2.42 2.42 0 0 1 1.773.732c.49.489.735 1.078.735 1.768s-.245 1.28-.735 1.768A2.42 2.42 0 0 1 2.492 21c-.692 0-1.28-.244-1.765-.732A2.418 2.418 0 0 1 0 18.5z"
          />
        </svg>
      </button>
    </div>
  </header>
  <div
    className="c10"
  >
    <div
      className="c11"
    >
      <div
        className="text text--dark text--primary c12"
        disabled={false}
        size={
          Object {
            "large": "uno",
            "medium": "uno",
            "small": "uno",
          }
        }
      >
        Henderson, NV â€” Sunset Station Outdoor Amphitheater
      </div>
      <div
        className="text text--dark text--secondary c13"
        disabled={false}
        size={
          Object {
            "large": "uno",
            "medium": "uno",
            "small": "uno",
          }
        }
      >
        On Sale: Fri, 08/17/18 10:00 AM
      </div>
    </div>
  </div>
  <footer
    className="c14"
  >
    <div
      className="c15"
    >
      <div
        className="c16"
      >
        <button
          className="c17"
          onClick={[Function]}
          size="small"
        >
          2:00 pm
        </button>
      </div>
      <div
        className="c16"
      >
        <button
          className="c17"
          onClick={[Function]}
          size="small"
        >
          6:00 pm
        </button>
      </div>
    </div>
    <div
      className="c18"
    >
      <button
        className="c19"
        onClick={[Function]}
        size="small"
      >
        +4 More Times
      </button>
    </div>
  </footer>
</article>
`;
